### PR TITLE
How to Make a Discord Bot using Python

### LINKED ISSUE #2424: How to Make a Discord Bot using Python
*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### How to Make a Discord Bot using Python

<br>

### Introduction
Discord is most used apps nowadays by companies, students and work professional and bots make it easy to protect our server from spams, greeting new people, verification , enjoyment and many more... .So I will give a full tutorial so that everyone can make their own bots using python for their servers, 

### Key takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article? Use this area to get your ideas down on the bulk of your article or tutorial.

1. Will learn how to make a bot using python
2. Add to the reader's skills to their most-using app and learn a bit about coding in python if not familiar yet.
3. How to use bot commands, checks, and converters.
4. How to create a Discord connection in Python.

### References: 
N/A

There is an article in EngEd about making a bot using node.js so I thought it would be helpful if one can also learn how to make it using python

### PR DESCRIPTION
**Pull request by harshitab13**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
This fixes #2424.
   
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by harshitab13** [comment]
This fixes #2744

**Comment by lalith1403** [comment]
@harshitab13 Welcome to the EngEd community. We are delighted to have you a part of our community. 

To publish your article, we will require a headshot pic and a short bio about yourself in the author's folder. For more details, check out the contribution guidelines and authors folder. 

Ping me in case you need more information.

Once again, welcome. Hope to see exceptional contributions on your end.

File: content/articles/how-to-make-a-discord-bot-using-python/index.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+This is fantastic, but it's only a example of how a bot may be beneficial. Once you learn how to build bots, there are a number of ways to get creative with them.
+
+**Note: Even though Discord permits you to develop bots for vocal communication, this tutorial will focus on the texting side of the platform.**
+
+When building a bot, there are two important stages to follow:
+1. Create a Discord bot user and assign it to a guild.
+2. Create code that executes your bot's actions and makes use of Discord's APIs.
+In the following part, you'll discover how to use Discord's Developer Portal to create a Discord bot.
+
+### OVERVIEW AND REQUIREMENTS
```
Comment (lalith1403) [inline content/articles/how-to-make-a-discord-bot-using-python/index.md:64]: Change all the titles. Only the first letter of the first word is capitalized. 

### Overview and requirements

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Upon a glance, there are quite a few grammatical errors. I suggest you fix them using online services such as Grammarly and then let us know.

**Comment by harshitab13** [comment]
@lalith1403 I have done all the fixes and added the author page

File: content/articles/how-to-make-a-discord-bot-using-python/index.md (hunk 1 above)
Comment (harshitab13) [reply content/articles/how-to-make-a-discord-bot-using-python/index.md]: Done

**Comment by ninjaginja** [comment]
plagiarism check completed - all good
[Uploading 2120420-30bf4378d8db3e5300ec-1626733611876.pdf…]()

File: content/articles/how-to-make-a-discord-bot-using-python/index.md [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+3.	A bot
+4.	A guild
+
+In the sections that precede, you'll discover much more about these components.
+
+When you've completed each of these components, you'll need to register your bot with your guild to connect everything.
+
+You may begin by entering [Discord's User Portal](https://discord.com/login?redirect_to=%2Fdevelopers%2Fapplications)
+
+1.	**Creating an Account** -
```
Comment (lalith1403) [inline content/articles/how-to-make-a-discord-bot-using-python/index.md:109]: Change the formatting to the H4 header. Remove the bold operator as well.

#### Creating an account

Let the first letter be capitalized, and the rest all start with small letters.

File: content/articles/how-to-make-a-discord-bot-using-python/index.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+Navigate to the "Bot" tab on the left-hand side panel, as this section is about how to build a Discord bot.
+
+3.	**Forming a Bot**
+
+A bot user listens to and automatically replies to specific events and orders on Discord, as discussed in the preceding parts.
+
+To see your code in action on Discord, you'll require to build a bot user. For that, select the "Add Bot" button.
+
+![bot](/engineering-education/how-to-make-a-discord-bot-using-python/img5.png)
```
Comment (lalith1403) [inline content/articles/how-to-make-a-discord-bot-using-python/index.md:149]: Change all the image URLs to the following format.

File: content/articles/how-to-make-a-discord-bot-using-python/index.md [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+![screen](./img8.png)
+
+Further, the final stage is to register the bot with the new guild at Discord.
+
+### Create a discord bot using discord.py library
+
+The bot's programming will be written in Python using the discord.py package. discord.py is a Discord API wrapper that makes creating a Discord bot in Python much more effortless.
+
+
+#### **Installing** `discord.py`  and creating a Repl
```
Comment (lalith1403) [inline content/articles/how-to-make-a-discord-bot-using-python/index.md:186]: Remove bold in titles. Use H3 and beyond to suitably identify the sections and sub-sections.

File: content/articles/how-to-make-a-discord-bot-using-python/index.md [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+
+```py -3 -m pip install -U discord.py```
+
+#### Setting up Discord event for your bot
+
+The structure of events is central to `discord.py`. An event is anything to which you pay attention and then respond. So, for example, when a message arrives, you will be notified via an event to which you can reply.
+
+Let's build a bot that responds to a given phrase. This simple bot code is derived from the discord.py manual, as is the [documentation](https://discordpy.readthedocs.io/en/latest/quickstart.html#a-minimal-bot). Later on, we'll add more functionality to the bot.
+
+This code should be added to main.py. (You may rename the file to whatever you like, don't call it discord.py.)
```
Comment (lalith1403) [inline content/articles/how-to-make-a-discord-bot-using-python/index.md:208]: Explaining why one shouldn't name it discord.py would definitely help. This is a problem many people encounter regularly and getting this knowledge in this article might help them elsewhere.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Hey @harshitab13. This is an amazing article. There are a few formatting issues that I have highlighted. Technically this article is sound—amazing job on the first article. 

Let me know once the changes are done.

**Comment by lalith1403** [comment]
Additionally, add the link to your repl in the conclusion section. That way, readers will have a resource to compare their work with.

**Comment by hectorkambow** [comment]
@harshitab13 

Just wanted to get a status update on this PR - so we can get in published as quickly as possible. Noticing that it has been in our queue for some time.

**Comment by harshitab13** [comment]
@lalith1403  done all the changes

**Comment by hectorkambow** [comment]
made some final edits for flow and format
Please be sure to review to see how future articles can be improved 

@harshitab13 
cc @lalith1403

**Pull request merged**